backfill.py

```diff
@@ -130,27 +146,165 @@ def get_result_from_durations(start_up_durations_path, analyzed_path):
+    apk_metadata_array = []
+
+    checkout_repository_to_correct_branch(repository_path, branch, remote_name)
+
+    if startdate is None:
+        array_of_commit_hash = get_all_commits_in_commits_range(start_commit, end_commit, repository_path)
+    else:
+        array_of_commit_hash = get_commits_for_date_range(startdate, enddate, repository_path)
+    for commit in array_of_commit_hash:
+        build_apk_for_commit(commit, repository_path, build_type)
+        built_apk_name = build_apk_path_string(repository_path, build_type, architecture)
+        new_apk_name = move_apk_to_cwd(built_apk_name, commit)
+        apk_metadata_array.append({
+            KEY_NAME: new_apk_name,
+            KEY_DATETIME: "",
+            KEY_COMMIT: commit,
+            KEY_ARCHITECTURE: architecture})
+    return apk_metadata_array
 
 
 def cleanup(array_of_apk_path):
     for i in array_of_apk_path:
         subprocess.run(["rm", i[KEY_NAME]])
 
 
+def validate_args(args):
+    if args.type == "commitsDate" and not args.startdate:
+        raise Exception("Running backfill with commits between two date requires a start date")
+    if args.type == "commitsRange" and not args.startcommit and not args.endcommit:
+        raise Exception("Running backfill with commits between two commits requires a start and end commit")
+    if (args.type == "commitsRange" or args.type == "commitsDate") and not args.repository_path:
+        raise Exception("Running backfill with any commits option " +
+                        "requires a path to a repository where git can be used")
+    if (args.type == "commitsRange" or args.type == "commitsDate") and not args.build_type:
+        raise Exception("Running backfill with any commits option requires a built type")
+
+
 def main():
     args = parse_args()
     if args.type == "commits" and args.fenix_path is None:
         raise Exception("Provide the path to your fenix repository to run this script with the commits option")
 
-    array_of_dates = get_date_array_for_range(args.start, args.enddate)
+    validate_args(args)
 
     if args.type == "nightly":
-        array_of_apk_metadata = download_nightly_for_range(array_of_dates)
-
-    run_performance_analysis_on_nightly(args.package_id, args.path_to_startup_script, array_of_apk_metadata)
+        array_of_dates = get_date_array_for_range(args.startdate, args.enddate)
+        array_of_apk_metadata = download_nightly_for_range(array_of_dates, args.architecture)
+    elif args.type == "commitsDate":
+        array_of_apk_metadata = build_apks_for_commits(
+            startdate=args.startdate,
+            enddate=args.enddate,
+            repository_path=args.repository_path,
+            build_type=args.build_type,
+            architecture=args.architecture,
+            branch=args.branch if args.branch else "",
+            remote_name=args.git_remote_name if args.git_remote_name else "")
+    else:
```

Comment:
nitpick (non-blocking): you should prefer being specific when possible. In this case, you should do, `elif args.type == 'commitsRange'` instead of `else`. This could be useful if someone adds a new `args.type` but forgets to update this if tree. With the current code, it'd do the wrong thing. With my suggestion, it'd crash at runtime.